src/ProcessManager.php

```diff
@@ -11,9 +11,10 @@ class ProcessManager
     /**
      * @param Process[] $processes
      * @param int $maxParallel
-     * @param int $poll
+     * @param int $poll microseconds
+     * @param Callable $callback takes 3 args: $type, $buffer, $process
      */
-    public function runParallel(array $processes, $maxParallel, $poll = 1000)
+    public function runParallel(array $processes, $maxParallel, $poll = 1000, $callback = null))
```

Comment:
Hello @gimler, thanks for your work. Your goal here is to "modify the command based on the previous finished". It looks like you are trying to accomplish a sequential process execution. This simple wrapper was meant for _parallel_ processing. Let me know if there is something that I am missing from my understanding.